packages/block-editor/src/components/block-settings-menu-controls/index.js

```diff
@@ -25,7 +25,12 @@ import { store as blockEditorStore } from '../../store';
 
 const { Fill, Slot } = createSlotFill( 'BlockSettingsMenuControls' );
 
-const BlockSettingsMenuControlsSlot = ( { fillProps, clientIds = null } ) => {
+const BlockSettingsMenuControlsSlot = ( {
+	fillProps,
+	clientIds = null,
+	selectedClientId,
```

Comment:
This new prop is critical. Previously, if you selected multiple blocks there was no way of accessing the _actual_ block that the menu was triggered on. The logic would only pass the single block clientId if a single block was passed.

I think that's an oversight but I'd appreciate a second pair of 👀

Reply:
When multiple blocks are selected, then I think only options that apply to multiple blocks should be visible.

This is already the case for some options like 'Edit as HTML' and 'Lock', they're both only visible for singular block selections.

Reply:
Looked into it some more, I'm fairly certain the existing `selectedClientIds` prop is all you need.

It works a bit like this:
- If a block selection is made, and the dropdown is triggered from a selected block, `selectedClientIds` contains the selected client ids (which can be one or more)
- If the dropdown is triggered from an unselected block, `selectedClientIds` contains only the client id of the block the menu was triggered on.

Reply:
Dan I missed this comment. I'm pretty sure there was a clear but that necessitated this but my comment is insufficient to explain the situation.

Essentially `selectedClientIds` gives you:

- the selected node
- the selected nodes (if multiple selected)

What it doesn't give you is the node who's 3 dots menu you clicked on.

So if I select a load of blocks and then click on the 3 dots menu on another node which _isn't selected_ the `selectedClientIds` gives me the nodes which are multi-selected and not the node which I expected.

I'll have to dig into it again to see if I can replicate and explain the situation in more detail.

Reply:
@talldan I tested this out and I couldn't replicate the issue I was originally experiencing.

Therefore I have

- reverted the addition of `selectedClientId` (singular) across all places
- enabled rename only for single block selections following pattern set by "Lock"